### PR TITLE
Bugfix/ses validation

### PR DESCRIPTION
**Pull request by justicel**
## what
* Adds a boolean flag to enable disabling creation of SPF/mx records and validation of SES domain

## why
* Not all users will want this module to manage their SES sending information (mx, spf)
* Not all users will need the domain to be validated by this module (in fact, I'd expect most won't)

### CODE REVIEW

File: outputs.tf [hunk 1]
```diff
@@ -1,11 +1,11 @@
 output "ses_domain_identity_arn" {
   description = "The ARN of the domain identity"
-  value       = aws_ses_domain_identity.default.arn
+  value       = var.domain_validation ? join("", aws_ses_domain_identity.default.*.arn) : null
```
Comment (Gowiem) [inline outputs.tf]: To stay consistent with our other modules, we typically just do the empty join pattern without the extra unneeded conditional. Can you update so value is just equal to `join("", aws_ses_domain_identity.default.*.arn)`. That will result in an empty string if `domain_validation` validation is false so accomplishes the same result.

File: outputs.tf [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 }
 
 output "ses_domain_identity_verification_arn" {
   description = "The ARN of the domain identity"
-  value       = aws_ses_domain_identity_verification.default.arn
+  value       = var.domain_validation ? join("", aws_ses_domain_identity_verification.default.*.arn) : null
```
Comment (Gowiem) [inline outputs.tf]: Same as above.

**Review by Gowiem — DISMISSED** [review]
One small request, but otherwise looks good. Update and we'll get this merged 👍

File: outputs.tf (hunk 2 above)
Comment (justicel) [reply outputs.tf]: Completed with latest change.

File: domain.tf [hunk 3]
```diff
@@ -1,19 +1,26 @@
...
+
 data "aws_route53_zone" "default" {
   name = var.domain
 }
 
 resource "aws_route53_record" "default" {
+  count   = local.enable_ses_validation
   zone_id = data.aws_route53_zone.default.zone_id
-  name    = "_amazonses.${aws_ses_domain_identity.default.id}"
+  name    = "_amazonses.${aws_ses_domain_identity.default.0.id}"
```
Comment (nitrocode) [inline domain.tf:13]: usually we use a join splat for each of these

could you change each one?

```suggestion
  name    = "_amazonses.${join("", aws_ses_domain_identity.default.*.id)}"
```

File: variables.tf [hunk 4]
```diff
@@ -18,6 +18,12 @@ variable "domain" {
   description = "Root domain name"
 }
 
+variable "domain_validation" {
```
Comment (nitrocode) [inline variables.tf:21]:
```suggestion
variable "domain_validation_enabled" {
```

File: variables.tf [hunk 5]
```diff
@@ -18,6 +18,12 @@ variable "domain" {
...
+  type        = bool
+  default     = true
+  description = "Boolean to enable or disable Route53 zone/domain validation in module"
```
Comment (nitrocode) [inline variables.tf:24]:
```suggestion
  description = "Whether or not to enable or disable Route53 zone/domain validation in module"
```

File: domain.tf [hunk 6]
```diff
@@ -1,19 +1,26 @@
+# Locals configuration to check if we should create SES validation or not (allow external management of this)
+locals {
+  enable_ses_validation = var.domain_validation ? 1 : 0
```
Comment (nitrocode) [inline domain.tf:3]:
```suggestion
  enable_ses_validation = var.domain_validation_enabled ? 1 : 0
```

**Comment by nitrocode** [comment]
@justicel last few changes. please see the comments

**Comment by phallows** [comment]
It would be good if this could be merged

**Pull request closed without merging**